lib/events.js

```diff
@@ -664,6 +641,38 @@ function createIterResult(value, done) {
   return { value, done };
 }
 
+function addErrorHandlerIfEventEmitter(emitter, handler, flags) {
```

Comment:
The flags should(?) eventually support all event target flags (like passive) where/if such extensions make sense. I don't like [boolean traps](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap) in general and I would prefer to allocate the object as its performance isn't measurable for the cost of just adding the listener compared to everything else going around like allocating the promise or async iterator.

Reply:
> I don't like boolean traps

Yeah, completely agree.

> eventually support all event target flags (like passive) 

I considered the 'eventually' here to be quite some time away and we could always just change it when we will really use it.
Anyway, there are many reasons to not have it as boolean that's why I commented it as just a Nit, so feel free to resolve and leave it as is.

Reply:
Ok, I'll resolve and to clarify I am hesitant because I am really not sure about a great many bunch of things here and your feedback is useful and appreciated.

I don't have a lot of strong opinions on this code base (yet), so if you do feel free to make them known and I will adjust the code in the PR :]